File: test/Models/energybalance_pkg/src/simplace/Ptsoil.java
import  java.io.*;
import  java.util.*;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import javafx.util.*;
import java.time.LocalDateTime;
import java.time.LocalDateTime;
import net.simplace.sim.model.FWSimComponent;
import net.simplace.sim.util.FWSimVarMap;
import net.simplace.sim.util.FWSimVariable;
import net.simplace.sim.util.FWSimVariable.CONTENT_TYPE;
import net.simplace.sim.util.FWSimVariable.DATA_TYPE;
import org.jdom.Element;


public class Ptsoil extends FWSimComponent
{
    private FWSimVariable<Double> Alpha;
    private FWSimVariable<Double> tau;
    private FWSimVariable<Double> tauAlpha;
    private FWSimVariable<Double> energyLimitedEvaporation;
    private FWSimVariable<Double> evapoTranspirationPriestlyTaylor;

    private Ptsoil(String aName, HashMap<String, FWSimVariable<?>> aFieldMap, HashMap<String, String> aInputMap, Element aSimComponentElement, FWSimVarMap aVarMap, int aOrderNumber)
    {
        super(aName, aFieldMap, aInputMap, aSimComponentElement, aVarMap, aOrderNumber);
    }

    public Ptsoil(){
        super();
    }
    @Override
    protected void process()
    {
        double tevapoTranspirationPriestlyTaylor = evapoTranspirationPriestlyTaylor.getValue();
        double tAlpha = Alpha.getValue();
        double ttau = tau.getValue();
        double ttauAlpha = tauAlpha.getValue();
        double tenergyLimitedEvaporation = energyLimitedEvaporation.getValue();
        double AlphaE;
        if (ttau < ttauAlpha)
        {
            AlphaE = 1.0d;
        }
        else
        {
            AlphaE = tAlpha - ((tAlpha - 1.0d) * (1.0d - ttau) / (1.0d - ttauAlpha));
        }
        tenergyLimitedEvaporation = tevapoTranspirationPriestlyTaylor / tAlpha * AlphaE * ttau;
        energyLimitedEvaporation.setValue(tenergyLimitedEvaporation, this);
    }

    @Override
    protected void init()
    {
    }

    @Override
    protected FWSimComponent clone(FWSimVarMap aVarMap)
    {
        return new Ptsoil(iName, iFieldMap, iInputMap, iSimComponentElement, aVarMap, iOrderNumber);
    }

    @Override
    public HashMap<String, FWSimVariable<?>> createVariables()
    {
        addVariable(FWSimVariable.createSimVariable("cAlpha", "Priestley-Taylor evapotranspiration proportionality constant", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 100, 1.5, this));
        addVariable(FWSimVariable.createSimVariable("ctau", "plant cover factor", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 100, 0.9983, this));
        addVariable(FWSimVariable.createSimVariable("ctauAlpha", "Fraction of the total net radiation exchanged at the soil surface when AlpaE = 1", DATA_TYPE.DOUBLE, CONTENT_TYPE.constant,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1, 0.3, this));
        addVariable(FWSimVariable.createSimVariable("energyLimitedEvaporation", "energy Limited Evaporation ", DATA_TYPE.DOUBLE, CONTENT_TYPE.out,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 5000, null, this));
        addVariable(FWSimVariable.createSimVariable("sevapoTranspirationPriestlyTaylor", "evapoTranspiration Priestly Taylor", DATA_TYPE.DOUBLE, CONTENT_TYPE.state,"http://www.wurvoc.org/vocabularies/om-1.8/", 0, 1000, 120, this));

        return iFieldMap;
    }
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SiriusComponent/sq_energybalance_generated/src/py/sq_energybalance_generated/Ptsoil.py
# coding: utf8
from copy import copy
from array import array
from math import *

import numpy

def model_ptsoil(evapoTranspirationPriestlyTaylor,
         Alpha,
         tau,
         tauAlpha):
    """
     - Name: PtSoil -Version: 001, -Time step: 1
     - Description:
                 * Title: PtSoil model
                 * Author: Peter D. Jamieson, Glen S. Francis, Derick R. Wilson, Robert J. Martin
                 * Reference: None
                 * Institution: New Zealand Institute for Crop and Food Research Ltd., New Zealand Institute for Crop and Food Research Ltd., New Zealand Institute for Crop and Food Research Ltd., New Zealand Institute for Crop and Food Research Ltd. 
                 * ExtendedDescription: Evaporation from the soil in the energy-limited stage 
                 * ShortDescription: None
     - inputs:
                 * name: evapoTranspirationPriestlyTaylor
                               ** description : evapoTranspiration of Priestly Taylor 
                               ** inputtype : variable
                               ** variablecategory : rate
                               ** datatype : DOUBLE
                               ** max : 10000
                               ** min : 0
                               ** default : 
                               ** unit : g m-2 d-1
                 * name: Alpha
                               ** description : Priestley-Taylor evapotranspiration proportionality constant
                               ** inputtype : parameter
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** max : 100
                               ** min : 0
                               ** default : 1.5
                               ** unit : 
                 * name: tau
                               ** description : plant cover factor
                               ** inputtype : parameter
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** max : 100
                               ** min : 0
                               ** default : 0.9983
                               ** unit : 
                 * name: tauAlpha
                               ** description : Fraction of the total net radiation exchanged at the soil surface when AlpaE = 1
                               ** inputtype : parameter
                               ** parametercategory : constant
                               ** datatype : DOUBLE
                               ** max : 1
                               ** min : 0
                               ** default : 0.3
                               ** unit : 
     - outputs:
                 * name: energyLimitedEvaporation
                               ** description : energy Limited Evaporation 
                               ** datatype : DOUBLE
                               ** variablecategory : auxiliary
                               ** max : 5000
                               ** min : 0
                               ** unit : g m-2 d-1
    """

    if tau < tauAlpha:
        AlphaE = 1.0
    else:
        AlphaE = Alpha - ((Alpha - 1.0) * (1.0 - tau) / (1.0 - tauAlpha))
    energyLimitedEvaporation = evapoTranspirationPriestlyTaylor / Alpha * AlphaE * tau
    return energyLimitedEvaporation